how-to/how-to-use-kata-containers-with-acrn.md

```diff
@@ -0,0 +1,126 @@
+
+To configure Docker for device-mapper and Kata, first, stop Docker daemon if it is already running using,
+
+```bash
+sudo systemctl stop docker
+```
+
+Set /etc/docker/daemon.json with the following contents:
+
+```bash
+{
+  "storage-driver": "devicemapper"
+}
+```
+
+Restart docker:
+
+```bash
+sudo systemctl daemon-reload
+sudo systemctl restart docker
+```
+
+Configure [Docker](https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md#update-the-docker-systemd-unit-file) to use kata-runtime
+
+## Configure Kata Containers with ACRN
+
+To configure Kata Containers with ACRN, all that is needed is to copy the configuration-acrn.toml file (please find attached) that is generated when building the kata-runtime to either /etc/kata-containers/configuration.toml or /usr/share/defaults/kata-containers/configuration.toml.
+
+Below command shows full paths to the configuration.toml files that the runtime loads.  The first path that exists is used. (Please make sure the kernel and image paths are set correctly in the configuration.toml file)
+
+```bash
+sudo kata-runtime --kata-show-default-config-paths
+```
+
+Once the configuration.toml is updated, please __offline the CPUs__ by creating a bash script using below and executing it to offline the CPUs.
+
+```bash
+echo "dm_run: before offline cpu" > /dev/kmsg
+# offline SOS CPUs except BSP before launch UOS
+for i in `ls -d /sys/devices/system/cpu/cpu[1-99]`; do
+        online=`cat $i/online`
+        idx=`echo $i | tr -cd "[1-99]"`
+        echo cpu$idx online=$online
+        if [ "$online" = "1" ]; then
+                echo 0 > $i/online
+                online=`cat $i/online`
+                # during boot time, cpu hotplug may be disabled
+                # by pci_device_probe during a pci module insmod
+                while [ "$online" = "1" ]; do
+                        sleep 0.1
+                        echo 0 > $i/online
+                        online=`cat $i/online`
+                done
+                echo $idx > /sys/class/vhm/acrn_vhm/offline_cpu
+        fi
+done
+echo "dm_run: after offline cpu" > /dev/kmsg
+```
+
+Start ACRN based kata container,
+
+```bash
+sudo docker run -ti --runtime=kata-runtime busybox sh
+```
+
+You'll see ACRN is now running on your system, as well as a kata-shim and kata-proxy processes: (make sure the shell launched is running, on exiting all the processes are terminated)
+
+```bash
+$ ps -ef | grep -E "kata|acrn"
+root      1574  1545  0 11:54 ?        00:00:00 /usr/bin/acrn-dm
+root      1620  1545  0 11:54 ?        00:00:00 /usr/libexec/kata-containers/kata-proxy
+root      1640  1545  0 11:54 pts/2    00:00:00 /usr/libexec/kata-containers/kata-shim
+```
+
+ACRN hypervisor can also be validated by using the kata-env command:
```

Comment:
`kata-env`

Reply:
I would move the `kata-env` in the configuration section above.

Reply:
Earlier configuration section is for Docker with Kata. So have not done this change.